app/Http/Controllers/VatsimOAuthController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericProvider;
+use League\OAuth2\Client\Token;
+
+class VatsimOAuthController extends GenericProvider {
+    /**
+     * @var GenericProvider
+     */
+    private $provider;
+
+    /**
+     * Initializes the provider variable.
+     */
+    public function __construct() {
+        parent::__construct([
+            'clientId'                => config('vatsim_auth.id'),    // The client ID assigned to you by the provider
+            'clientSecret'            => config('vatsim_auth.secret'),   // The client password assigned to you by the provider
+            'redirectUri'             => route('login'),
+            'urlAuthorize'            => config('vatsim_auth.base').'/oauth/authorize',
+            'urlAccessToken'          => config('vatsim_auth.base').'/oauth/token',
+            'urlResourceOwnerDetails' => config('vatsim_auth.base').'/api/user',
+            'scopes'                  => config('vatsim_auth.scopes'),
+            'scopeSeparator'          => ' '
+        ]);
+    }
+
+    /**
+     * Gets an (updated) user token
+     * @param Token $token
+     * @return Token
+     * @return null
+     */
+    public static function updateToken($token) {
+        $controller = new VatsimOAuthController;
+        try {
+            return $controller->getAccessToken('refresh_token', [
+                'refresh_token' => $token->getRefreshToken()
+            ]);
+        } catch (IdentityProviderException $e) {
+            return null;
+        }
+    }
```

Comment:
I think this can be removed as well?

Reply:
Removed this function - no longer needed since we aren't using the refresh token for anything